recipe/meta.yaml

```diff
@@ -44,7 +38,7 @@ requirements:
     - pybind11
     - scikit-build >=0.11.0
     - setuptools >=40.1.0
-    - spdlog =1.5.0
+    - spdlog =1.9.2
```

Comment:
Q: Do we need to use this exact version, or is it OK to use the globally pinned version?
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/c854f3a2bdeb645121552d466537490158d2bcdf/recipe/conda_build_config.yaml#L716-L717

Reply:
@wshanks CI is green now, so this is my only remaining question 🙂

Reply:
I think you have a better sense of this than I do. I was just going by what qiskit-aer puts [here](https://github.com/Qiskit/qiskit-aer/blob/6f600adcf2c4621fab79ff00000a933e43cd2941/cmake/dependency_utils.cmake#L27) in its cmake configuration. I think the configuration in the qiskit-aer repo is focused mostly on producing something that is known to work rather than something that works best in a larger ecosystem. Would you recommend removing the version entirely as long as the build succeeds? Should we also do that for nlohmann_json?

Reply:
Understood. Let's give it a shot then, as I've seen subtle `spdlog` issues before. I'll change it.

As for `nlohmann_json`, it's not pinned globally, so let's not worry about it.

Reply:
@wshanks, all green, shall we merge?